llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp

```diff
@@ -61,6 +61,79 @@ AArch64LegalizerInfo::AArch64LegalizerInfo(const AArch64Subtarget &ST)
   const LLT v2s64 = LLT::fixed_vector(2, 64);
   const LLT v2p0 = LLT::fixed_vector(2, p0);
 
+  // Scalable vector sizes range from 128 to 2048
+  // Note that subtargets may not support the full range.
+  // See [ScalableVecTypes] below.
+  const LLT nxv16s8 = LLT::scalable_vector(16, s8);
+  const LLT nxv32s8 = LLT::scalable_vector(32, s8);
+  const LLT nxv64s8 = LLT::scalable_vector(64, s8);
+  const LLT nxv128s8 = LLT::scalable_vector(128, s8);
+  const LLT nxv256s8 = LLT::scalable_vector(256, s8);
```

Comment:
This is a scalable vector, which size is an unknown multiple of 256 * 8. Legalizing loads of scalable vectors is unintuitive.

Reply:
As far as I know, there is no target that supports `nxv256s8`. The original patch only listed:
```
 const LLT nxv8s16 = LLT::scalable_vector(8, 16);
 const LLT nxv4s32 = LLT::scalable_vector(4, 32);
 const LLT nxv2s64 = LLT::scalable_vector(2, 64);
```

Reply:
To make it simpler for now, I would remove all these except for base sizes of 128-bit:
- `nxv8s16`, `nxv4s32`, and `nxv2s64` as listed in the original patch
- `nxv16s8` since SelectionDAG supports it
- `nxv2p0` since it has the same effective size as `nxv2s64`.